src/main/java/com/ssomar/score/features/custom/cooldowns/CooldownsManager.java

```diff
@@ -27,16 +38,71 @@ public static CooldownsManager getInstance() {
+                    // Extract the slot number using regex
+                    String slotNumber = params.replaceFirst(".*cooldown_slot(\\d+).*", "$1");
+                    int slot = Integer.parseInt(slotNumber);
+
+                    // Retrieve equipment for the slot
+                    if (player.isOnline() && player.getPlayer() != null) {
+                        ItemStack item = player.getPlayer().getInventory().getItem(slot);                   
+                        if (item != null) {
+                            if (item.hasItemMeta() && player != null) {
+                                ItemMeta iM = item.getItemMeta();
+            
+                                NamespacedKey key = new NamespacedKey(ExecutableItems.getPluginSt(), "ei-id");
+                                String eiid = iM.getPersistentDataContainer().get(key, PersistentDataType.STRING);
+            
+                                if (eiid != null) {
+                                    cooldownId += "EI:" + eiid + ":";
+                                    // get rid of the slot
+                                    params = params.replaceFirst("slot\\d+:", "");
+                                } else {
+                                    return Optional.of("Not an Executable Item: " + item.getItemMeta().getDisplayName());
+                                }
+                            } else {
+                                return Optional.of("No ItemMeta found or player is null");
+                            }
+                        } else {
+                            return Optional.of("No Item in specified slot");
+                        }
+                    } else {
+                        return Optional.of("Player Offline");
+                    }
+
+                }
+
+                // Player requested int, toggle and remove from query.
                 if(params.contains("_int")){
                     castInt = true;
                     params = params.replaceFirst("_int", "");
                 }
+                if(params.contains("_timemax")) {
+                    castMax = true;
+                    params = params.replaceFirst("_timemax", "");                    
+                }
+
+                // Parse as normal, minor refactor so it's easier to read
+                cooldownId += params.split("cooldown_")[1];
                 for(Cooldown cd : cooldownsUUID.get(uuid)){
                     SsomarDev.testMsg("CD "+cd.toString(), DEBUG);
-                    if(cd.getId().equalsIgnoreCase(params.split("cooldown_")[1])){
-                        if(castInt) return Optional.of((int)cd.getTimeLeft()+"");
+                    if(cd.getId().equalsIgnoreCase(cooldownId)){
+
+                        // Return result if found based on booleans above 
+                        if(castInt && castMax) return Optional.of((int)cd.getCooldown()+"");
+                        else if(castMax) return Optional.of(cd.getCooldown()+"");
```

Comment:
Be conscious that it didnt return anymore the max if the player disconnected.
It will return the time left at his re-connection

Reply:
I don't know how to fix this :(